### PR TITLE
Handle protection status errors for unprotected branches

### LINKED ISSUE #625: Branch protection status checks returns error for unprotected branches
Response is 404 so throws an error, instead of returning nil slice of required contexts. Sending in a quick PR.

### PR DESCRIPTION
**Pull request by hemachandarv**
Resolves #625

### CODE REVIEW

**Comment by hemachandarv** [comment]
@gmlewis I've addressed #625 by returning an explicit error in the `GET` APIs. Do you think this is a good idea?

Also, do you think we should cover the same for the `POST` APIs too?

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @hemachandarv !
Just a few minor tweaks, please.

File: github/repos_test.go
Comment (hemachandarv) [reply github/repos_test.go:1162]: @gmlewis I'm not sure if we'll be able to use a variable inside a raw string. Am I missing something..?

File: github/repos.go
Comment (hemachandarv) [reply github/repos.go:1563]: @gmlewis Thank you for suggesting this change. I certainly missed this simplification.

File: github/repos_test.go
Comment (gmlewis) [reply github/repos_test.go:1162]: Whups, you are correct. Please use `%v` and the variable name in that case.

Comment (hemachandarv) [reply github/repos_test.go]: Thanks @gmlewis. I've updated the PR with the suggested changes.

**Review by gmlewis — APPROVED** [review]
Thank you, @hemachandarv !
LGTM.

Awaiting second LGTM before merging.

**Review by cpanato — APPROVED** [review]
LGTM
thanks!

**Comment by gmlewis** [comment]
Thank you, @cpanato !
Merging.

**Pull request merged**
